src/Main.elm.d.ts

```diff
@@ -1,40 +1,42 @@
-export namespace Elm {
-  namespace Main {
-    interface App {
-      ports: Ports;
-    }
+declare module '*.elm' {
```

Comment:
- Needed since "type": "module",
- https://discourse.elm-lang.org/t/how-to-use-elm-in-tsx-file/4556/3